packages/core/core/src/requests/PathRequest.js

```diff
@@ -1,32 +1,67 @@
-// @flow
-
-import type {AssetRequestDesc, Dependency, ParcelOptions} from './types';
+// @flow strict-local
 import type {Diagnostic} from '@parcel/diagnostic';
-import type ParcelConfig from './ParcelConfig';
+import type ParcelConfig from '../ParcelConfig';
+import type {StaticRunOpts} from '../RequestTracker';
+import type {AssetGroup, Dependency, ParcelOptions} from '../types';
 
-import {PluginLogger} from '@parcel/logger';
-import {relatifyPath} from '@parcel/utils';
 import ThrowableDiagnostic, {errorToDiagnostic} from '@parcel/diagnostic';
+import {PluginLogger} from '@parcel/logger';
+import {escapeMarkdown, relatifyPath} from '@parcel/utils';
 import path from 'path';
 import URL from 'url';
+import {report} from '../ReporterRunner';
+import PublicDependency from '../public/Dependency';
+import PluginOptions from '../public/PluginOptions';
+
+export type PathRequestInput = {|dependency: Dependency, config: ParcelConfig|};
+export type PathRequestResult = AssetGroup | null | void;
+
+export type PathRequest = {|
+  id: string,
+  +type: 'path_request',
+  run: RunOpts => Promise<PathRequestResult>,
+  input: PathRequestInput,
+|};
+
+type RunOpts = {|
+  input: PathRequestInput,
+  ...StaticRunOpts,
+|};
 
-import {report} from './ReporterRunner';
-import PublicDependency from './public/Dependency';
-import PluginOptions from './public/PluginOptions';
+const type = 'path_request';
 
-import {escapeMarkdown} from '@parcel/utils';
+export default function createPathRequest(input: PathRequestInput) {
+  return {
+    id: input.dependency.id,
+    type,
+    run,
+    input,
+  };
+}
+async function run({input, api, options}: RunOpts) {
+  let {config, dependency} = input;
+  let resolverRunner = new ResolverRunner({options, config});
+  let assetGroup = await resolverRunner.resolve(dependency);
+
+  // ? Should this happen if asset is deferred?
```

Comment:
You mean resolving or invalidating?

Reply:
This is a question I asked a while ago that accidentally slipped through. Pretty sure I was wondering about invalidation. Now that I have more understanding I around that, I can answer that the invalidation should still be set up. Even if the file was never transformed, if it's removed we will want to resolve again. I'll remove the comment.